python/cuml/dask/ensemble/base.py

```diff
@@ -288,13 +316,59 @@ def get_combined_model(self):
 
         return internal_model
 
+    def apply_reduction(self, reduce, partial_infs, datatype, delayed):

```

Comment:
A comment explaining the flow here and why weighting is necessary would be helpful